paasta_tools/kubernetes_tools.py

```diff
@@ -1966,6 +1966,13 @@ def sanitize_for_config_hash(
             service=self.get_service(), environment_variables=self.get_env()
         )
 
+        # Only record whether a boto_keys hash exists
+        # This allows the service to restart once when boto_keys secret is
+        # first added, not when the values change
+        if ahash.get("boto_keys"):
+            boto_secret_hash = self.get_boto_secret_hash()
+            ahash["boto_keys_sig"] = "present" if boto_secret_hash else ""
```

Comment:
doesn't this end up being:
```suggestion
        if ahash.get("boto_keys") and self.get_boto_secret_hash()
            ahash["boto_keys_sig"] = "present"
```
? `get_boto_secret_hash` is typed as always returning a `str`, so i don't think we'll ever get anything but `"present"` if we're in this block

and just to confirm the behavior: we want to bounce a service when we go from having 0 -> N keys or N -> 0 keys - but not from N -> N ± 1 keys?